src/main/java/mate/academy/springboot/web/service/impl/BookServiceImpl.java

```diff
@@ -38,4 +38,28 @@ public BookDto findById(Long id) {
     public BookDto create(CreateBookRequestDto requestDto) {
         return bookMapper.toDto(bookRepository.save(bookMapper.toModel(requestDto)));
     }
+
+    @Override
+    public BookDto updateBook(Long id, BookDto bookDto) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(String.format(BOOK_NOT_FOUND, id)));
+
+        book.setTitle(bookDto.getTitle());
+        book.setAuthor(bookDto.getAuthor());
+        book.setPrice(bookDto.getPrice());
+        book.setIsbn(bookDto.getIsbn());
+        book.setDescription(bookDto.getDescription());
+        book.setCoverImage(bookDto.getCoverImage());
+
+        Book updatedBook = bookRepository.save(book);
+        return bookMapper.toDto(updatedBook);
+    }
+
+    @Override
+    public void deleteBook(Long id) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(String.format(BOOK_NOT_FOUND, id)));
+        book.setIsDeleted(true);
+        bookRepository.save(book);
```

Comment:
also you can check if book exists
```suggestion
        bookRepository.deleteById(id);
```